runtime/queries/metricsview_timeseries.go

```diff
@@ -72,73 +71,53 @@ func (q *MetricsViewTimeSeries) Resolve(ctx context.Context, rt *runtime.Runtime
 		return fmt.Errorf("metrics view '%s' does not have a time dimension", q.MetricsViewName)
 	}
 
-	// Build query
-	sql, args, err := q.buildMetricsTimeSeriesSQL(mv)
+	measures, err := toMeasures(mv.Measures, q.MeasureNames)
 	if err != nil {
-		return fmt.Errorf("error building query: %w", err)
+		return err
 	}
 
-	// Execute
-	meta, data, err := metricsQuery(ctx, olap, priority, sql, args)
+	tsq := &ColumnTimeseries{
+		TableName:           mv.Model,
+		TimestampColumnName: mv.TimeDimension,
+		TimeRange: &runtimev1.TimeSeriesTimeRange{
+			Start:    q.TimeStart,
+			End:      q.TimeEnd,
+			Interval: q.TimeGranularity,
+		},
+		Measures: measures,
+		Filters:  q.Filter,
+	}
+	err = rt.Query(ctx, instanceID, tsq, priority)
 	if err != nil {
 		return err
 	}
 
+	r := tsq.Result
+
 	q.Result = &runtimev1.MetricsViewTimeSeriesResponse{
-		Meta: meta,
-		Data: data,
+		Meta: r.Meta,
+		Data: r.Data.Results,
```

Comment:
Is `r.Data` guaranteed to never be `nil`?

Reply:
Also, since we're not using all the data returned here, have we ensured this doesn't run more queries than necessary?

Reply:
If you mean the Spark query, it is not run if `Pixels == 0`